src/core/components/table/CoachList.vue

```diff
@@ -171,11 +171,11 @@ export default {
       return this.roles.map(role => ({ label: ROLES_TRANSLATION[role.name], value: role._id }));
     },
     canSetUserCompany () {
-      const ability = defineAbilitiesFor(
-        get(this.loggedUser, 'role.client.name'),
-        get(this.loggedUser, 'role.vendor.name'),
-        this.company
-      );
+      const ability = defineAbilitiesFor({
+        clientRole: get(this.loggedUser, 'role.client.name'),
+        vendorRole: get(this.loggedUser, 'role.vendor.name'),
+        company: this.company,
+      });
```

Comment:
ici tu peux directement passer `pick(this.loggedUser, ['_id', role, company]`

d'ailleurs la company qui est envoyée n'est pas celle de l'utilisateur mais celle de la structure qu'on regarde donc a changer